### PR TITLE
slurm: use modern meson build syntax and use newer ncurses

### PR DESCRIPTION
**Pull request by chenrui333**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

### CODE REVIEW

File: Formula/s/slurm.rb [hunk 1]
```diff
@@ -20,15 +20,12 @@ class Slurm < Formula
 
   depends_on "meson" => :build
   depends_on "ninja" => :build
-
-  uses_from_macos "ncurses"
+  depends_on "ncurses"
```
Comment (SMillerDev) [inline Formula/s/slurm.rb:23]: Why?

Comment (p-linnane) [reply Formula/s/slurm.rb:23]: The Sonoma builds failed without that change. Everything else was successful.

Comment (SMillerDev) [reply Formula/s/slurm.rb:23]: Can we make it depend on Sonoma then?

Comment (p-linnane) [reply Formula/s/slurm.rb:23]: FWIW, I wasn't able to reproduce this locally on Sonoma. But scoping to just Sonoma makes sense if we are moving forward.

Comment (chenrui333) [reply Formula/s/slurm.rb:23]: yeah, we can just scope this change ot sonoma builds, but I think the `ncurses 5` has lots of security bugs.

Comment (chenrui333) [reply Formula/s/slurm.rb]: @p-linnane I think it might be because of you already had `ncurses 6` installed?

Comment (chenrui333) [reply Formula/s/slurm.rb]: actually uninstalled `ncurses` also worked for me with a sonoma-arm machine. weird.

Comment (SMillerDev) [reply Formula/s/slurm.rb]: >  but I think the ncurses 5 has lots of security bugs.

macOS doesn't ship with version 5, at least not on Ventura.

**Pull request closed without merging**